Parser/Python.asdl

```diff
@@ -76,7 +76,7 @@ module Python
          -- x < 4 < 3 and (x < 4) < 3
          | Compare(expr left, cmpop* ops, expr* comparators)
          | Call(expr func, expr* args, keyword* keywords)
-         | FormattedValue(expr value, int? conversion, expr? format_spec)
+         | FormattedValue(expr value, int? conversion, expr? format_spec, string? expr_text)
```

Comment:
It is not needed and I prefer to not add redundant attributes. Instead you can add the Constant item just before FormattedValue into JoinedStr.

Reply:
It's used in ast_unparse.c to know if the = needs to be added back in. It's not strictly necessary, of course. It depends how you feel about the fidelity required in string annotations.

Reply:
I think it if fine if `f'{foo=}'` be transformed into `f'foo={foo!r}'`. Think about this as a kind of macros.

Reply:
I discussed this very point with Łukasz yesterday. I think his point is that he'd like to not lose the information in the unparser. Admittedly it's not perfect, and it probably can't ever be without a ton of work and additions to the AST.